core/src/libtx/build.rs

```diff
@@ -199,47 +199,58 @@ where
 }
 
 /// Builds a complete transaction.
+/// NOTE: We only use this in tests (for convenience).
+/// In the real world we use signature aggregation across multiple participants.
 pub fn transaction<K, B>(
```

Comment:
This function isn't used anywhere yet, right?

Reply:
No this is used extensively in our tests. Search for `build::transaction`.
It is also used in some tests in grin-wallet, but I think just legacy from when the repos were originally split (those tests should probably live in grin itself).

Reply:
The new part of this is where we call `build::transaction_with_kernel()` internally from `build::transaction()` allowing for more flexibility.